src/path_calculator.rs

```diff
@@ -173,7 +173,10 @@ impl PathCalculator {
             let adj = graph.edges_bwd[edge_id].adj_node;
             let edge_weight = graph.edges_bwd[edge_id].weight;
             let adj_weight = self.get_weight_fwd(adj);
-            if adj_weight != WEIGHT_MAX && adj_weight + edge_weight < curr.weight {
+            if graph.ranks[adj] > graph.ranks[curr.node_id]
```

Comment:
I still do not see why we need to check the ranks here. Do you have an example where no path is found without this check as you said here: https://github.com/easbar/fast_paths/issues/18#issuecomment-613560398?